module-bluetooth/Bluetooth/BluetoothWorker.cpp

```diff
@@ -156,15 +157,16 @@ auto BluetoothWorker::handleCommand(QueueHandle_t queue) -> bool
         break;
     case bluetooth::Command::Unpair:
         controller->processCommand(command);
-        removeFromBoundDevices(command.getDevice().address);
-        handleUnpairDisconnect(command.getDevice());
+        removeFromBoundDevices(std::get<Devicei>(command.getData()).address);
+        handleUnpairDisconnect(std::get<Devicei>(command.getData()));
         break;
     case bluetooth::Command::None:
         break;
     default:
         controller->processCommand(command);
         break;
     }
+    command.cleanup();
```

Comment:
I can do cleanup while executing the `getData()` method, but then I don't have any control of the data when the user doesn't retrieve the data from the command. What would you suggest @pholat @swichu91

Reply:
cleaning up before function exit is not bad in here. This is very c like - and handleCommand is very c style function :)

I would think of a way to remove the switch/case needed here. If we could move it somewhere else for parsing? Then we can unit test parsing :) (then use data in designated handlers after it's parsed) especially that there is generic `else: I don't know what I do in switch case -> controler->procesCommand(command)`

Reply:
Do we need to manage underlying data manually? We can store `std::shared_ptr` as a `Command` member instead of a raw pointer. By doing so, memory will be managed automatically. The only potential problem I see here is raw FreeRTOS queue doesn't take ownership of the data it contains.

Reply:
@swichu91 that's how it's made - the queue makes the problem as after copying the previous owner makes the data out of scope, so it's getting deleted. I tried with `shared_ptr` and I've always received `nullptr`.

@pholat I think that's possible, but I did this command refactor to be able to finish the task - I didn't want to rewrite whole BT with this job :P